pkg/consensus/consensus.go

```diff
@@ -299,6 +298,7 @@ func (s *service) Shutdown() {
 }
 
 func (s *service) eventLoop() {
+	s.Chain.SubscribeForBlocks(s.blockEvents)
```

Comment:
So we need the following code to be added before the `events` loop handling to fix the race described in https://github.com/nspcc-dev/neo-go/pull/3307#issuecomment-1966309470:
```
anna@kiwi:~/Documents/GitProjects/nspcc-dev/neo-go$ git diff
diff --git a/pkg/consensus/consensus.go b/pkg/consensus/consensus.go
index 5eea615e..fcedfef0 100644
--- a/pkg/consensus/consensus.go
+++ b/pkg/consensus/consensus.go
@@ -300,6 +300,14 @@ func (s *service) Shutdown() {
 
 func (s *service) eventLoop() {
        s.Chain.SubscribeForBlocks(s.blockEvents)
+
+       // Manually sync up with potentially missed fresh blocks that may be added by blockchain
+       // before the subscription.
+       b, _ := s.Chain.GetBlock(s.Chain.CurrentBlockHash()) // Can't fail, we have some current block!
+       if b.Timestamp >= s.lastTimestamp {
+               s.handleChainBlock(b)
+       }
+
 events:
        for {
                select {
```

@roman-khimov, agree to add such change?

Reply:
Makes sense.

Reply:
Not critical for other services, only consensus is affected.